lib/cheffish/base_resource.rb

```diff
@@ -0,0 +1,241 @@
+require 'chef_compat/resource'
+
+module Cheffish
+  class BaseResource < ChefCompat::Resource
+    declare_action_class.class_eval do
```

Comment:
`action_class` only works if there is already an action class. `declare_action_class` will make sure there is one. An action class will already be there if you are a subclass of something with an action class, or if you declare `action`.